src/protocol.h

```diff
@@ -277,6 +277,35 @@ enum ServiceFlags : uint64_t {
     // BIP process.
 };
 
+/**
+ * Gets the set of service flags which are "relevant" for a given peer.
+ *
+ * Generally, avoid calling with peerServices == NODE_NONE as we may have
+ * service flags which are optionally relevant (eg NODE_NETWORK_LIMITED).
+ *
+ * If the peer is missing any of the returned service flags, they aren't
+ * interesting to us - we probably shouldn't conenct outbound to them.
+ */
+static ServiceFlags GetRelevantServiceFlags(ServiceFlags services) {
```

Comment:
Can you document what the parameter represents? Hard to tell since it's unused. I assume it's for future compatibility.

Reply:
Indeed, its for use in #10387. Will it be clearer when the function reads
```GetRelevantServiceFlags(services) {
    if (services & NODE_NETWORK_LIMITED && !in_ibd) return NODE_NETWORK_LIMITED | NODE_WITNESS;
    return NODE_NETWORK | NODE_WITNESS;
``` or should I document it further?

Reply:
Yeah, parameter documentation would be nice because I still find its purpose confusing. Especially since as a parameter of `HasAllRelevantServiceFlags` it represents both the service flags you want to test and a modifier on what you want to test against. Like, why is the future code not just:

```c++
GetRelevantServiceFlags() {
    if (!in_ibd) return NODE_NETWORK_LIMITED | NODE_WITNESS;
    else return NODE_NETWORK | NODE_WITNESS;
```

Reply:
You need the NODE_NETWORK_LIMITED check as a proxy for whether the node is new - eg if the node announces NODE_NETWORK_LIMITED (and you're out of IBD) then you're fine with NODE_NETWORK_LIMITED, but if the node only announces NODE_NETWORK they may be an 0.15.X node and you're perfectly OK with them having only NODE_NETWORK and not NODE_NETWORK_LIMITED.

Reply:
Oh, ok, I figured that would work because NODE_NETWORK would be a superset of NODE_NETWORK_LIMITED. But if NODE_NETWORK without NODE_NETWORK_LIMITED is a meaningful flag combination then it makes sense that you need the services argument.

In any case, I mostly wanted to suggest the renames here.

Reply:
NODE_NETWORK without NODE_NETWORK_LIMITED is not meaningful generally, but old nodes will (obviously) be using that, so it is important that we still treat it correctly.